airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt

```diff
@@ -109,18 +113,123 @@ class SnowflakeSqlGenerator(private val retentionPeriodDays: Int) : SqlGenerator
         minRawTimestamp: Optional<Instant>,
         useExpensiveSaferCasting: Boolean
     ): Sql {
-        val insertNewRecords =
-            insertNewRecords(stream, finalSuffix, minRawTimestamp, useExpensiveSaferCasting)
-        var dedupFinalTable = ""
-        var cdcDeletes = ""
+
         if (stream.destinationSyncMode == DestinationSyncMode.APPEND_DEDUP) {
-            dedupFinalTable =
-                dedupFinalTable(stream.id, finalSuffix, stream.primaryKey, stream.cursor)
-            cdcDeletes = cdcDeletes(stream, finalSuffix)
+            return transactionally(
+                upsertRecords(stream, finalSuffix, minRawTimestamp, useExpensiveSaferCasting),
+                checkpointRawTable(stream.id)
+            )
         }
-        val commitRawTable = commitRawTable(stream.id)
+        return insertRecordsNoDedupe(stream, finalSuffix, minRawTimestamp, useExpensiveSaferCasting)
+    }
+
+    // TODO: Delete this or keep it behind a flag to fallback if MERGE doesn't work out.
+    private fun insertRecordsDedupe(
```

Comment:
If both methods exist, should this be an advanced options that users can toggle?  Let's get a lot of data here... we already rolled this out and back once before

Reply:
yeah. thats what i was leaning towards just to avoid yet another rollback for any unforeseen reason.

Reply:
Added a flag in spec to toggle the behavior.